Module4.3_Search_and_Sorting_Algorithms/activities/Leet Code Activity/3.md

```diff
@@ -0,0 +1,11 @@
+Now that we have our linked list definition, we can start on the actual solution.The first part of the code is :
+
+```python
+def insertionSortList(self, head):
+    p = dummy = ListNode(0)
+    cur = dummy.next = head
+```
+
+In this part of the code, p is initialized along with dummy to have a val of 0 and be the root of the linked list. In our solution, p's role will be to be the pointer for where cur will insert after.  And our dummy variable is the linked list that we will return as a sorted linked list. 
+
+In the second line of the code, cur and dummy.next are initialized to be the head of the linked list. In our solution, cur will be using to compare with other elements. 
```

Comment:
"will be using"?

can u give a reminder as to why we have to use pointers?

also, in a general/overview sense, what is the function of insertSortList()? (pun intended)